bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/ThreadFactoryBuilder.java

```diff
@@ -0,0 +1,201 @@
+ * used, e.g., when creating {@link Executor}s via the {@link Executors} utility methods.
+ * <p/>
+ * The built {@link ThreadFactory} uses a wrapped {@link ThreadFactory} to create threads (defaulting to
+ * {@link Executors#defaultThreadFactory()}, and then overwrites thread properties as indicated in the build process.
+ *
+ * @author Henning Sudbrock - initial contribution
+ */
+@NonNullByDefault
+public class ThreadFactoryBuilder {
+
+    @Nullable
+    private ThreadFactory wrappedThreadFactory = null;
+    @Nullable
+    private String namePrefix = null;
+    private boolean createDaemonThreads = false;
+    @Nullable
+    private UncaughtExceptionHandler uncaughtExceptionHandler = null;
+    @Nullable
+    private Integer priority = null;
+
+    /**
+     * Creates a fresh {@link ThreadFactoryBuilder}.
+     *
+     * @return A {@link ThreadFactoryBuilder}
+     */
+    public static ThreadFactoryBuilder create() {
+        return new ThreadFactoryBuilder();
+    }
+
+    private ThreadFactoryBuilder() {
+        // use static factory method to create ThreadFactoryBuilder
+    }
+
+    /**
+     * Sets the wrapped thread factory used to create threads. If set to null, {@link Executors#defaultThreadFactory()}
+     * is used. Defaults to null.
+     *
+     * @param wrappedThreadFactory the wrapped thread factory to be used
+     * @return this {@link ThreadFactoryBuilder} instance
+     */
+    public ThreadFactoryBuilder withWrappedThreadFactory(@Nullable ThreadFactory wrappedThreadFactory) {
+        this.wrappedThreadFactory = wrappedThreadFactory;
+        return this;
+    }
+
+    /**
+     * Sets the name prefix to be used by the {@link ThreadFactory}.
+     * <p/>
+     * The threads created by the {@link ThreadFactory} will be named 'ESH-namePrefix-i', where i is an integer
+     * incremented with each new thread, initialized to 1.
+     * <p/>
+     * If the namePrefix is null, the naming strategy from the wrapped {@link ThreadFactory} is used. Defaults to null.
+     *
+     * @param namePrefix The name prefix (can be null)
+     * @return this {@link ThreadFactoryBuilder} instance
+     */
+    public ThreadFactoryBuilder withNamePrefix(@Nullable String namePrefix) {
+        this.namePrefix = namePrefix;
+        return this;
+    }
+
+    /**
+     * Sets whether the {@link ThreadFactory} will create daemon threads. Defaults to false.
+     *
+     * @param daemonThreads indicates whether daemon threads shall be created
+     * @return this {@link ThreadFactoryBuilder} instance
+     */
+    public ThreadFactoryBuilder withDaemonThreads(boolean daemonThreads) {
+        this.createDaemonThreads = daemonThreads;
+        return this;
+    }
+
+    /**
+     * Sets the {@link ThreadFactory} to create daemon threads.
+     *
+     * @return this {@link ThreadFactoryBuilder} instance
+     */
+    public ThreadFactoryBuilder withDaemonThreads() {
```

Comment:
Hm, AFAIK its more common to use a boolean argument instead of adding two methods (one for true and one for false).

Reply:
Hm, I added both here (i.e., three methods in total; one method with Boolean argument, and two methods for use without argument). The reason is that I have met people that like the Boolean argument variant, and others that prefer using the non-argument variant - both arguing that it would make the code more readable.

I would not find for any of both variants; do you think the PR should be updated by removing the two no-argument variants?